service/src/test/java/tds/assessment/web/endpoints/ItemControllerIntegrationTests.java

```diff
@@ -0,0 +1,99 @@
+package tds.assessment.web.endpoints;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.util.Optional;
+
+import tds.assessment.ItemFileMetadata;
+import tds.assessment.ItemFileType;
+import tds.assessment.services.ItemService;
+import tds.common.configuration.JacksonObjectMapperConfiguration;
+import tds.common.configuration.SecurityConfiguration;
+import tds.common.web.advice.ExceptionAdvice;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(ItemController.class)
+@Import({ExceptionAdvice.class, SecurityConfiguration.class, JacksonObjectMapperConfiguration.class})
+public class ItemControllerIntegrationTests {
+    @MockBean
+    private ItemService mockItemService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private MockMvc http;
+
+    @Test
+    public void shouldReturnStimulusItemMetadata() throws Exception {
+        String url = "/assessments/item/metadata?clientName=SBAC_PT&bankKey=1&stimulusKey=3";
+
+        ItemFileMetadata itemFileMetadata = ItemFileMetadata.create(ItemFileType.STIMULUS, "1-3", "stimulusFile", "stimulusFile/");
+        when(mockItemService.findItemFileMetadataByStimulusKey("SBAC_PT", 1, 3)).thenReturn(Optional.of(itemFileMetadata));
+
+        MvcResult result = http.perform(get(url)
```

Comment:
This is a different way of doing our controller integration tests.  As @burzum619 commented last week it is a bit of a pain that we don't verify that the client objects can be parsed by Jackson.  So I figured I'd try to see if we could tweak our integration test style to accommodate that complaint, and I think this works pretty well.

1. Inject the ObjectMapper that is created `JacksonObjectMapperConfiguration`
2. Use the `MockMvc` like always except instead of running assertions against the JSON body content return the `MvcResult`.
3. Use the objectMapper to read the content value (from `MvcResult.getResponse().getContentAsByteArray()`.
4. Do assertions on the translated object.

The `ObjectMapper` injected is the same one used by the RestTemplate in our other services.  This should reduce the number of times we run into the json parsing errors though of course if the caller doesn't configure the object mapper appropriately it'll still blow up but I see no way around that.